Example/Tests/BONHumanReadableStringTestCase.m

```diff
@@ -0,0 +1,85 @@
+    BONChain *imageChain = BONChain.new.image(barnImage);
+    BONChain *textChain = BONChain.new.string(@"concatenate me!");
+
+    [imageChain appendLink:textChain];
+
+    BONAssertEquivalentStrings(imageChain.attributedString, @"{image36x36}concatenate me!")
+}
+
+- (void)testSpecialCharacters
+{
+    BONChain *chain = BONChain.new.string(@"This");
+    [chain appendLink:BONChain.new.string(@"string") separator:BONSpecial.tab];
+    [chain appendLink:BONChain.new.string(@"is") separator:BONSpecial.lineFeed];
+    [chain appendLink:BONChain.new.string(@"populated") separator:BONSpecial.space];
+    [chain appendLink:BONChain.new.string(@"by") separator:BONSpecial.noBreakSpace];
+    [chain appendLink:BONChain.new.string(@"BONSpecial") separator:BONSpecial.enSpace];
+    [chain appendLink:BONChain.new.string(@"characters") separator:BONSpecial.figureSpace];
+    [chain appendLink:BONChain.new.string(@"that") separator:BONSpecial.thinSpace];
+    [chain appendLink:BONChain.new.string(@"will") separator:BONSpecial.hairSpace];
+    [chain appendLink:BONChain.new.string(@"be") separator:BONSpecial.zeroWidthSpace];
+    [chain appendLink:BONChain.new.string(@"replaced") separator:BONSpecial.nonBreakingHyphen];
+    [chain appendLink:BONChain.new.string(@"by") separator:BONSpecial.figureDash];
+    [chain appendLink:BONChain.new.string(@"simple") separator:BONSpecial.enDash];
+    [chain appendLink:BONChain.new.string(@"human") separator:BONSpecial.emDash];
+    [chain appendLink:BONChain.new.string(@"readable") separator:BONSpecial.horizontalEllipsis];
+    [chain appendLink:BONChain.new.string(@"versions") separator:BONSpecial.lineSeparator];
+    [chain appendLink:BONChain.new.string(@"of") separator:BONSpecial.paragraphSeparator];
+    [chain appendLink:BONChain.new.string(@"their") separator:BONSpecial.narrowNoBreakSpace];
+    [chain appendLink:BONChain.new.string(@"own") separator:BONSpecial.wordJoiner];
+    [chain appendLink:BONChain.new.string(@"name") separator:BONSpecial.minusSign];
+
+    NSString *expectedHumanReadableString = @"This{tab}string{lineFeed}is populated{noBreakSpace}by{enSpace}BONSpecial{figureSpace}characters{thinSpace}that{hairSpace}will{zeroWidthSpace}be{nonBreakingHyphen}replaced{figureDash}by{enDash}simple{emDash}human{horizontalEllipsis}readable{lineSeparator}versions{paragraphSeparator}of{narrowNoBreakSpace}their{wordJoiner}own{minusSign}name";
+    NSAttributedString *testAttributedStringProperty = chain.attributedString;
+
+    BONAssertEquivalentStrings(testAttributedStringProperty, expectedHumanReadableString);
+}
+
+- (void)testAllTheCharactersInTheKitchenSink
+{
+    BONChain *everything = BONChain.new;
+    [everything appendLink:BONChain.new.string(@"neon")];
+    [everything appendLink:BONChain.new.string(@"سلام")];
+    [everything appendLink:BONChain.new.string(@"🚲")];
+    [everything appendLink:BONChain.new.string(@"\U000A1337") separator:BONSpecial.figureDash];
```

Comment:
#nth/#? this character is good for the other test where it's used, but it actually reveals a problem with this string: it appears to be non-printing in this case, and its official name is `\N{<unassigned-A1337>}`. Is there any way to categorically detect that a character like this, and make it display in the human-readable string? If not, we should at least file a bug and use a different character for this test that makes the intent more clear. Or, leave this character in, so that the test fails if we ever fix it.